### PR TITLE
feat: add inter-chunk-latency metric using raw value list

### PR DESCRIPTION
**Pull request by ajcasagrande**
- Inter Chunk Latency measures the time gaps between consecutive response chunks in a streaming response. 
- For each request with multiple response chunks, this metric computes the time difference between each adjacent pair of chunks.

Example:
For a request starting at 0ms, with streaming response chunks arriving at times:
`responses = [100ms, 150ms, 210ms] -> ICL = [50ms, 60ms]`
    
All ICL values from all requests are collected into a single list for statistical analysis across the entire benchmark run.

>[!Tip]
>Formula:
> ```python
> ICL = [timestamps[i] - timestamps[i-1] for i in range(1, len(timestamps))]
> ```
    
Note: Only applicable to streaming responses with at least 2 chunks.

_Experimental metric only available in dev mode._

Screenshot From 2025-09-23 13-34-20

## Summary by CodeRabbit

- New Features
  - Added Inter-Chunk Latency metric for streaming responses, reporting per-chunk latencies (displayed in milliseconds).
  - Added support for list-valued metrics (lists of integers and floats) across the metrics pipeline.

- Refactor
  - Metric storage now accepts batches of values via an extend operation and auto-resizes for larger datasets.
  - Added validation for initial metric storage capacity.

- Tests
  - Added comprehensive tests for the new latency metric and metric storage behaviors, covering correctness, edge cases, and error handling.

### CODE REVIEW

File: aiperf/post_processors/metric_results_processor.py [hunk 1]
```diff
@@ -75,7 +75,12 @@ async def process_result(self, incoming_metrics: MetricRecordDict) -> None:
                 if metric_type == MetricType.RECORD:
                     if tag not in self._results:
                         self._results[tag] = MetricArray()
-                    self._results[tag].append(value)  # type: ignore
+                    if isinstance(value, list):
+                        # NOTE: Right now we only support list-based metrics by extending the array.
+                        #       In the future, we possibly could support having nested arrays.
+                        self._results[tag].extend(value)  # type: ignore
+                    else:
+                        self._results[tag].append(value)  # type: ignore
```
[the remark this answers is not in the document]
Comment (ajcasagrande) [reply aiperf/post_processors/metric_results_processor.py:83]: not seeing what this is talking about

File: aiperf/metrics/metric_dicts.py [hunk 2]
```diff
@@ -92,19 +92,31 @@ def __init__(self, initial_capacity: int = 10000):
...
+        self._resize_if_needed(1)
 
         self._data[self._size] = value
         self._size += 1
         self._sum += value  # type: ignore
 
+    def _resize_if_needed(self, additional_size: int) -> None:
+        """Resize the array if needed."""
+        if self._size + additional_size > self._capacity:
+            self._capacity *= 2
```
[the remark this answers is not in the document]
Comment (ajcasagrande) [reply aiperf/metrics/metric_dicts.py]: done

File: aiperf/common/enums/metric_enums.py [hunk 3]
```diff
@@ -329,6 +329,18 @@ class MetricValueType(BasePydanticBackedStrEnum):
...
+        default_factory=list[float],
+        converter=list[float],
+        dtype=list[float],
+    )
+    INT_LIST = MetricValueTypeInfo(
+        tag="list[int]",
+        default_factory=list[int],
+        converter=list[int],
+        dtype=list[int],
+    )
```
[the remark this answers is not in the document]
Comment (ajcasagrande) [reply aiperf/common/enums/metric_enums.py:343]: the top part has been resolved. the bottom part I find no reason for change, as the current code works
